ImageD11/frelon_peaksearch.py

```diff
@@ -339,11 +339,14 @@ def masksub(self, img):
             img_masked[self.mask == 1] = 0
         return img_masked
 
-    def peaksearch(self, img, omega=0):
+    def peaksearch(self, img, omega=0, scale_factor=None):
         if self.wrk is None:
             self.wrk = np.empty(img.shape, "b")
             self.labels = np.empty(img.shape, "i")
-
+        if scale_factor is not None:
```

Comment:
You need to pass the scale_factor into bgsub and apply it after self.correct(img).